src/main/java/org/broadinstitute/dsde/consent/ontology/datause/services/TextTranslationServiceImpl.java

```diff
@@ -89,13 +98,65 @@ public String translateDataset(String dataUseString) throws IOException {
         return translate(dataUse, TranslateFor.DATASET);
     }
 
+    @Override
+    public HashMap<String, Recommendation> translateParagraph(String paragraph) throws Exception {
+        return paragraph(paragraph);
+    }
+
     @Override
     public String translatePurpose(String dataUseString) throws IOException {
         ObjectMapper mapper = new ObjectMapper();
         DataUse dataUse = mapper.readValue(dataUseString, DataUse.class);
         return translate(dataUse, TranslateFor.PURPOSE);
     }
 
+    private HashMap<String, Recommendation> paragraph(final String paragraph) throws Exception {
+      HashMap<String, Recommendation> recommendations = new HashMap<>();
+
+      List<TermItem> terms = loadTermsFromGoogleStorage();
+
+      for (TermItem term : terms) {
+        final String title = term.getTitle();
+        final String category = term.getCategory();
+        final String url = term.getUrl();
+        final String[] keywords = term.getKeywords();
+
+        for (String keyword : keywords) {
+          final boolean foundMatch = searchForKeyword(keyword, paragraph);
+
+          if (foundMatch) {
+            Recommendation recommendation = new Recommendation(
+                title,
+                category
+            );
+            if (!recommendations.containsKey(url)) {
+              recommendations.put(url, recommendation);
+            }
+          }
+        }
+      }
+
+      return recommendations;
+    }
+
+    private static boolean searchForKeyword(final String keyword, final String targetText) {
+      return StringUtils.containsIgnoreCase(targetText, keyword);
+    }
+
+    List<TermItem> loadTermsFromGoogleStorage() throws Exception {
+      HttpResponse response = gcsStore.getStorageDocument("ontology/search-terms.json");
+      String terms = response.parseAsString();
```

Comment:
These two methods throw an IOException. If we hit those, we really should log them as errors. I think we could move these into the try/catch and deal with the different exceptions more explicitly. Lastly, if we did hit an error deserializing `search-terms.json`, we should log that as well since it is a different category of error.